File: server/src/com/broll/gainea/server/core/fractions/FractionDescription.java
package com.broll.gainea.server.core.fractions;

import java.util.ArrayList;
import java.util.List;

public class FractionDescription {

    private String general;
    private List<String> plus = new ArrayList<>();
    private List<String> contra = new ArrayList<>();

    public FractionDescription(String general) {
        this.general = general;
    }

    public void plus(String strength) {
        plus.add(strength);
    }

    public void contra(String weakness) {
        contra.add(weakness);
    }

    public List<String> getContra() {
        return contra;
    }

    public List<String> getPlus() {
        return plus;
    }

    public String getGeneral() {
        return general;
    }
}

File: server/src/main/com/broll/gainea/server/core/fractions/FractionDescription.kt
package com.broll.gainea.server.core.fractions

class FractionDescription(val general: String, val soldier: UnitDescription, val commander: UnitDescription) {
    private val plus = mutableListOf<String>()
    private val contra = mutableListOf<String>()
    operator fun plus(strength: String) {
        plus.add(strength)
    }

    fun contra(weakness: String) {
        contra.add(weakness)
    }

    fun getContra(): List<String> {
        return contra
    }

    fun getPlus(): List<String> {
        return plus
    }
}
